frontend-react/src/hooks/UseLookupTable.ts

```diff
@@ -91,14 +90,54 @@ export const getSenderAutomationData = async <T>(
     );
 };
 
-const useLookupTable = <T>(tableName: LookupTables): ValueSet[] => {
-    const [valueSetArray, setValueSetArray] = useState<ValueSet[]>([]);
+export const getSenderAutomationDataRows = async <T>(
+    tableName: LookupTables,
+    dataSetName: string | null = null
+): Promise<any[]> => {
+    const version: number = await getLatestVersion(tableName);
+    if (version === undefined) {
+        showError("DANGER! no version was found");
+        return [];
+    }
+    const data: T | any[] = await getLatestData<T[]>(version, tableName);
+
+    return data
+        .filter((f) => f.name === dataSetName)
+        .map(
+            (set: {
+                display: string;
+                code: string;
+                version: string;
+                system: string;
+            }) => ({
+                display: set.display,
+                code: set.code,
+                version: set.version,
+                system: set.system,
+            })
+        );
+};
+
+const useLookupTable = <T>(
+    tableName: LookupTables,
+    dataSetName: string | null = null
+): T[] => {
```

Comment:
I _hope_ I can use the work I'm doing in #5557 to refactor the fetching hooks later. Looks like it'll be a good stress test!